### PR TITLE
Add base_game & level_1_1

### PR DESCRIPTION
**Pull request by novayo**
https://github.com/novayo/Mario_cplus/pull/23 這個確認好沒問題後再進

Issue: https://github.com/novayo/Mario_cplus/issues/16

### CODE REVIEW

File: level/base_level.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <unordered_map>
```
Comment (tkwang0530) [inline level/base_level.h:1]:
```suggestion
#ifndef BASE_LEVEL_H
#define BASE_LEVEL_H
#include <unordered_map>
```

File: level/base_level.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+typedef std::tuple<int, int> POS; 
+typedef std::tuple<int, int, int> COLOR; 
+
+class BaseLevel {
+public:
+    virtual void play_background_music();
+    virtual void stop_background_music();
+    virtual COLOR get_background_color();
+    virtual std::unordered_map<POS, std::string> get_backgrounds();
+};
```
Comment (tkwang0530) [inline level/base_level.h:13]:
```suggestion
};

#endif // BASE_LEVEL_H
```

File: level/base_level.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+typedef std::tuple<int, int> POS; 
+typedef std::tuple<int, int, int> COLOR; 
+
+class BaseLevel {
+public:
+    virtual void play_background_music();
+    virtual void stop_background_music();
+    virtual COLOR get_background_color();
+    virtual std::unordered_map<POS, std::string> get_backgrounds();
```
Comment (tkwang0530) [inline level/base_level.h:12]:
```suggestion
    virtual const std::unordered_map<POS, std::string> get_backgrounds();
```

Comment (tkwang0530) [reply level/base_level.h:12]: 使用 const 引用可以避免不必要的copy

File: level/base_level.h (hunk 1 above)
Comment (novayo) [reply level/base_level.h]: Done.

File: level/base_level.h (hunk 2 above)
Comment (novayo) [reply level/base_level.h:16]: Done.

File: level/base_level.h (hunk 3 above)
Comment (novayo) [reply level/base_level.h]: Done.

**Review by tkwang0530 — APPROVED** [review]

**Pull request merged**